### PR TITLE
Scan compile/grad speed up

### PR DESCRIPTION
**Pull request by nouiz**
When the inner of scan is big, this should give good speed up.

We remove useless clone in the L_op and R_op. It also remove useless clone during the Scan.__init__() method. So this should also speed up opt that create new Scan node.

### CODE REVIEW

**Comment by nouiz** [comment]
jenkins test this

**Comment by nouiz** [comment]
I rebased to have the CI run on the current master to be sure all is good. If they pass, I'll merge as it was approuved before the rebase.

File: theano/scan_module/scan_op.py [hunk 1]
```diff
@@ -207,10 +207,10 @@ def tensorConstructor(broadcastable, dtype):
         if self.info['gpu'] or self.info['gpua']:
             self._hash_inner_graph = self.info['gpu_hash']
         else:
-            tmp_in, tmp_out = scan_utils.reconstruct_graph(self.inputs,
-                                                           self.outputs)
-            # This is actually required for the line just after.
-            gof.FunctionGraph(tmp_in, tmp_out, clone=False)
+            # gof don't know about SharedVariable. So add them as inputs.
```
Comment (abergeron) [inline theano/scan_module/scan_op.py]: This comment refers to SharedVariable, but the code below deals with constants.

File: theano/scan_module/tests/test_scan.py [hunk 2]
```diff
@@ -4488,8 +4488,8 @@ def _scan_loose(x):
                                n_steps=n,
                                strict=True)
 
-        f_strict = theano.function([x0_], ret_strict[0][-1])
-        result_strict = f_strict(x0)
+#        f_strict = theano.function([x0_], ret_strict[0][-1])
+#        result_strict = f_strict(x0)
```
Comment (abergeron) [inline theano/scan_module/tests/test_scan.py]: Remove, don't comment.

Comment (nouiz) [inline theano/scan_module/tests/test_scan.py]: both done.

**Pull request merged**